### PR TITLE
chore(doc): Add using cloudinary plugin for images to reference guides. 

### PR DESCRIPTION
**Pull request by Ekwuno**
## Description
This PR addresses https://app.clubhouse.io/gatsbyjs/story/3122/create-reference-guide-on-using-the-cloudinary-image-service

This suggests an option to use the cloudinary plugins as an alternative to gatsby-image. 

### Documentation
https://www.gatsbyjs.org/docs/images-and-files/

## Related Issues
N/A

### CODE REVIEW

File: docs/docs/using-cloudinary-image-service.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
+---
+title: Using cloudinary image service for media optimization
```
Comment (laurieontech) [inline docs/docs/using-cloudinary-image-service.md:2]:
```suggestion
title: Using Cloudinary image service for media optimization
```

File: docs/docs/using-cloudinary-image-service.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+---
+
+`Cloudinary` is a cloud-based end-to-end media management platform for many of the world’s top brands. With extensive product offerings from an Image management solution, Dynamic Video Platform, and a dynamic digital asset management solution.
+
+As Plugins are at the heart of developing software with Gatsby and optimization is also really important in mordern app. The Cloudinary plugins aid in making sure media assets fit this need.
```
Comment (laurieontech) [inline docs/docs/using-cloudinary-image-service.md:7]:
```suggestion
As plugins are at the heart of developing software with Gatsby and optimization is also really important in mordern app. The Cloudinary plugins aid in making sure media assets fit this need.
```
This needs rewording. The first sentence is a bit confusing.

File: docs/docs/using-cloudinary-image-service.md [hunk 3]
```diff
@@ -0,0 +1,148 @@
+---
+title: Using cloudinary image service for media optimization
+---
+
+`Cloudinary` is a cloud-based end-to-end media management platform for many of the world’s top brands. With extensive product offerings from an Image management solution, Dynamic Video Platform, and a dynamic digital asset management solution.
```
Comment (laurieontech) [inline docs/docs/using-cloudinary-image-service.md:5]: Are all of these offerings relevant to Gatsby sites?

File: docs/docs/using-cloudinary-image-service.md [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+
+As Plugins are at the heart of developing software with Gatsby and optimization is also really important in mordern app. The Cloudinary plugins aid in making sure media assets fit this need.
+
+In this guide you will take a look at the [gatsby-source-plugin](/packages/gatsby-source-cloudinary/) and [gatsby-transformer-cloudinary](/packages/gatsby-transformer-cloudinary/) which can be used to improve the experience of handling images on Gatsby sites.
```
Comment (laurieontech) [inline docs/docs/using-cloudinary-image-service.md:9]:
```suggestion
In this guide you will take a look at [gatsby-source-plugin](/packages/gatsby-source-cloudinary/) and [gatsby-transformer-cloudinary](/packages/gatsby-transformer-cloudinary/) which can be used to improve the experience of handling images on Gatsby sites.
```

File: docs/docs/using-cloudinary-image-service.md [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+
+Here's a [Demo site for source plugin](https://gsc-sample.netlify.com) showcasing optimized images in a masonry grid, served from [Cloudinary](https://cloudinary.com)
```
Comment (laurieontech) [inline docs/docs/using-cloudinary-image-service.md:11]:
```suggestion
Here's a [demo site for source plugins](https://gsc-sample.netlify.com), showcasing optimized images in a masonry grid, served from [Cloudinary](https://cloudinary.com)
```

Comment (laurieontech) [reply docs/docs/using-cloudinary-image-service.md:11]: We should link to Cloudinary up top at first mention. Then we don't need to do it again.

File: docs/docs/using-cloudinary-image-service.md (hunk 3 above)
Comment (laurieontech) [inline docs/docs/using-cloudinary-image-service.md:5]:
```suggestion
Cloudinary is a cloud-based end-to-end media management platform for many of the world’s top brands. With extensive product offerings from an Image management solution, Dynamic Video Platform, and a dynamic digital asset management solution.
```

File: docs/docs/using-cloudinary-image-service.md [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+
+The `gatsby-source-plugin` is helpful if you already have some images hosted on `Cloudinary` and would like to leverage on the plugin system of Gatsby.
```
Comment (laurieontech) [inline docs/docs/using-cloudinary-image-service.md:13]:
```suggestion
The `gatsby-source-plugin` is helpful if you already have some images hosted on Cloudinary and would like to leverage on the plugin system of Gatsby.
```

Comment (laurieontech) [reply docs/docs/using-cloudinary-image-service.md:13]: Can you explain what you mean? Is there another way to query this information? "like to leverage the plugin system" feels misleading. Perhaps it's more, if you'd like to access your images via Cloudinary.

File: docs/docs/using-cloudinary-image-service.md [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+
+## Problem
+
+Dealing with images on the web have always been a problem as unoptimized images can slow down your site. The processes put in place to create the best media experience can take a lot of time to implement.
+
+## Solutions
+
+Cloudinary provides a couple of amazing solutions to this problem by offering the following solutions:
```
Comment (laurieontech) [inline docs/docs/using-cloudinary-image-service.md:21]:
```suggestion
Cloudinary provides a couple of amazing solutions to this problem, namely:
```

File: docs/docs/using-cloudinary-image-service.md [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+
+## Gatsby-Source-plugin
+
+This plugin fetches media assets from Cloudinary that are specified in a folder. It then transforms these images into Cloudinary file nodes, which can be queried with GraphQL in a Gatsby project.
+
+`gatsby-source-plugin` applies [f_auto and q_auto](https://cloudinary.com/documentation/image_transformations) transformation parameters which aid in automatic optimisation of format and quality for media assets by over 70 percent.
+
+### Prerequisites
+
+There are a couple of things you would need to set up before using this plugin:
```
Comment (laurieontech) [inline docs/docs/using-cloudinary-image-service.md:35]:
```suggestion
```

File: docs/docs/using-cloudinary-image-service.md [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+
+### Usage
+
+1. Install `gatsby-source-cloudinary`
+
+```shell
+  npm install gatsby-source-cloudinary
+```
+
+2. In the root of your project, create an environment file called `.env` to which to add your Cloudinary credentials and their values
```
Comment (laurieontech) [inline docs/docs/using-cloudinary-image-service.md:49]:
```suggestion
2. In the root of your project, create an environment file called `.env` and add your Cloudinary credentials and their values
```

File: docs/docs/using-cloudinary-image-service.md [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+```
+CLOUDINARY_API_KEY=xxxxxxxxxxxxxx
+CLOUDINARY_API_SECRET=xxxxxxxxxxxxxxxxxxxx
+CLOUDINARY_CLOUD_NAME=xxxxx
+```
+
+3. Configure `gatsby-config.js`
+
+```js:title=gatsby-config.js
+require('dotenv').config();
```
Comment (laurieontech) [inline docs/docs/using-cloudinary-image-service.md:60]: Highlight this line perhaps?

File: docs/docs/using-cloudinary-image-service.md [hunk 11]
```diff
@@ -0,0 +1,148 @@
...
+- **`resourceType`** **:** This is the resource type of the media assets: either an image or a video.
+- **`prefix`** **:** This is the folder (in your Cloudinary account) in which the files reside. In the example above, I named this folder `gatsby-source-cloudinary` . Assign a name of your choice.
+
+Other optional options are `type`, `tags`, and `maxResult`.
+
+> The `info` log, which displays the CloudinaryMedia nodes. Those images are ready to be queried in Gatsby components.
+
+## Gatsby-transformer-cloudinary
+
+Here's a [Demo site for transformer plugin](https://gatsby-transformer-cloudinary.netlify.com/fluid/)
```
Comment (laurieontech) [inline docs/docs/using-cloudinary-image-service.md:91]:
```suggestion
Here's a [demo site for transformer plugin](https://gatsby-transformer-cloudinary.netlify.com/fluid/)
```

File: docs/docs/using-cloudinary-image-service.md [hunk 12]
```diff
@@ -0,0 +1,141 @@
...
+
+## The problem with handling images on the web
+
+Dealing with images on the web has always been a problem as unoptimized images can slow down your site. The processes put in place to create the best media experience can take a lot of time to implement. 
+
+## Solutions Cloudinary provides 
+
+Cloudinary provides a couple of amazing solutions to this problem , namely: 
+- Remote storage and delivery of images via CDN
+- Wider range of transformations over [gatsby-image](/docs/using-gatsby-image/).
```
Comment (laurieontech) [inline docs/docs/using-cloudinary-image-service.md:22]:
```suggestion
- Wider range of transformations over [gatsby-image](/docs/using-gatsby-image/).
```
What do we mean by over?

File: docs/docs/using-cloudinary-image-service.md [hunk 13]
```diff
@@ -0,0 +1,141 @@
...
+- [Digital Asset Management](https://cloudinary.com/documentation/digital_asset_management_overview) for enterprise assets 
+
+## Gatsby-source-cloudinary
+
+This plugin fetches media assets from Cloudinary that are specified in a folder. It then transforms these images into Cloudinary file nodes, which can be queried with GraphQL in a Gatsby project. 
+`gatsby-source-cloudinary` applies [f_auto and q_auto](https://cloudinary.com/documentation/image_transformations) transformation parameters which aid in automatic optimisation of format and quality for media assets by over 70 percent.
+
+### Prerequisites  
+
+Before using the `gatsby-source-cloudinary` you should do the following :
```
Comment (laurieontech) [inline docs/docs/using-cloudinary-image-service.md:32]:
```suggestion
Before using the `gatsby-source-cloudinary` plugin you should do the following: 
```

File: docs/docs/using-cloudinary-image-service.md [hunk 14]
```diff
@@ -0,0 +1,141 @@
...
+## Gatsby-source-cloudinary
+
+This plugin fetches media assets from Cloudinary that are specified in a folder. It then transforms these images into Cloudinary file nodes, which can be queried with GraphQL in a Gatsby project. 
+`gatsby-source-cloudinary` applies [f_auto and q_auto](https://cloudinary.com/documentation/image_transformations) transformation parameters which aid in automatic optimisation of format and quality for media assets by over 70 percent.
+
+### Prerequisites  
+
+Before using the `gatsby-source-cloudinary` you should do the following : 
+
+- Upload the images to a folder on Cloudinary. This folder can have any name of your choosing.
```
Comment (laurieontech) [inline docs/docs/using-cloudinary-image-service.md:34]:
```suggestion
- Upload your images to a folder on Cloudinary. This folder can have any name of your choosing. 
```

File: docs/docs/using-cloudinary-image-service.md [hunk 15]
```diff
@@ -0,0 +1,141 @@
...
+This plugin fetches media assets from Cloudinary that are specified in a folder. It then transforms these images into Cloudinary file nodes, which can be queried with GraphQL in a Gatsby project. 
+`gatsby-source-cloudinary` applies [f_auto and q_auto](https://cloudinary.com/documentation/image_transformations) transformation parameters which aid in automatic optimisation of format and quality for media assets by over 70 percent.
+
+### Prerequisites  
+
+Before using the `gatsby-source-cloudinary` you should do the following : 
+
+- Upload the images to a folder on Cloudinary. This folder can have any name of your choosing. 
+- Obtain your API key and API secret from your Cloudinary dashboard.
+- Have [dotenv](https://www.npmjs.com/package/dotenv) module installed for loading environment variables from a `.env` file.
```
Comment (laurieontech) [inline docs/docs/using-cloudinary-image-service.md:36]:
```suggestion
- Have the [dotenv](https://www.npmjs.com/package/dotenv) module installed for loading environment variables from a `.env` file.
```

File: docs/docs/using-cloudinary-image-service.md [hunk 16]
```diff
@@ -0,0 +1,141 @@
...
+
+### Using gatsby-source-cloudinary
+
+Add the plugin to your project in these steps :
```
Comment (laurieontech) [inline docs/docs/using-cloudinary-image-service.md:40]:
```suggestion
Add the plugin to your project.
```

File: docs/docs/using-cloudinary-image-service.md [hunk 17]
```diff
@@ -0,0 +1,141 @@
...
+        apiKey: process.env.CLOUDINARY_API_KEY,
+        apiSecret: process.env.CLOUDINARY_API_SECRET,
+        resourceType: `image`,
+        prefix: `gatsby-source-cloudinary/` 
+      }
+    }
+  ]
+} 
+```
+> `gatsby-source-cloudinary` takes these options:
```
Comment (laurieontech) [inline docs/docs/using-cloudinary-image-service.md:76]:
```suggestion
Note that `gatsby-source-cloudinary` takes the following options:
```

File: docs/docs/using-cloudinary-image-service.md [hunk 18]
```diff
@@ -0,0 +1,141 @@
...
+
+- **`cloudName`** , **`apiKey`** , and **`apiSecret`** **:** These are credentials from your Cloudinary console, stored as three separate environment variables for security.
+- **`resourceType`** **:** This is the resource type of the media assets: either an image or a video.
```
Comment (laurieontech) [inline docs/docs/using-cloudinary-image-service.md:79]:
```suggestion
- **`resourceType`** **:** This is the resource type of the media assets - either an image or a video.
```

File: docs/docs/using-cloudinary-image-service.md [hunk 19]
```diff
@@ -0,0 +1,141 @@
...
+- **`prefix`** **:** This is the folder (in your Cloudinary account) in which the files reside. In the example above, I named this folder `gatsby-source-cloudinary` . Assign a name of your choice.
```
Comment (laurieontech) [inline docs/docs/using-cloudinary-image-service.md:80]:
```suggestion
- **`prefix`** **:** This is the folder (in your Cloudinary account) in which the files reside. In the example above, the folder is called `gatsby-source-cloudinary`. Assign a name of your choice.
```

File: docs/docs/using-cloudinary-image-service.md [hunk 20]
```diff
@@ -0,0 +1,141 @@
...
+Other optional options are `type`, `tags`, and `maxResult`.
```
Comment (laurieontech) [inline docs/docs/using-cloudinary-image-service.md:81]: Maybe link to the README if there is more context there?

File: docs/docs/using-cloudinary-image-service.md [hunk 21]
```diff
@@ -0,0 +1,141 @@
...
+
+> The `info` log, which displays the Cloudinary Media nodes. Those images are ready to be queried in Gatsby components.
```
Comment (laurieontech) [inline docs/docs/using-cloudinary-image-service.md:83]: Not sure what the context for this statement is. Is this another option?

File: docs/docs/using-cloudinary-image-service.md [hunk 22]
```diff
@@ -0,0 +1,141 @@
...
+
+## Gatsby-transformer-cloudinary
+
+After sourcing media files from Cloudinary, you will be able to leverage on Cloudinary’s media transformation capabilities you will make use of the `gatsby-transformer-cloudinary` which is a type of [transformer plugin](/docs/creating-a-transformer-plugin/) that is used to change image formats , style and dimension. While also optimizes images for minimal file size alongside high visual quality for an improved user experience and minimal bandwidth.
```
Comment (laurieontech) [inline docs/docs/using-cloudinary-image-service.md:87]:
```suggestion
After sourcing media files from Cloudinary, you will be able to leverage Cloudinary’s media transformation capabilities. To do so,  use `gatsby-transformer-cloudinary` which is a type of [transformer plugin](/docs/creating-a-transformer-plugin/) that is used to change image formats, styles and dimensions. It also optimizes images for minimal file size alongside high visual quality for an improved user experience and minimal bandwidth.
```
The last sentence is a bit confusing. Can we clarify?

File: docs/docs/using-cloudinary-image-service.md [hunk 23]
```diff
@@ -0,0 +1,141 @@
...
+
+Here's a [demo site that uses the gatsby-transformer-plugin](https://gatsby-transformer-cloudinary.netlify.com/fluid/)
+
+### Using gatsby-transformer-cloudinary
+
+1. Install `gatsby-transformer-cloudinary` and  `gatsby-source-filesystem` which creates the File nodes that the Cloudinary transformer plugin works on. 
+
+```shell
+ npm install --save gatsby-transformer-cloudinary gatsby-source-filesystem
```
Comment (laurieontech) [inline docs/docs/using-cloudinary-image-service.md:96]:
```suggestion
 npm install gatsby-transformer-cloudinary gatsby-source-filesystem 
```

File: docs/docs/using-cloudinary-image-service.md [hunk 24]
```diff
@@ -0,0 +1,141 @@
...
+Here's a [demo site that uses the gatsby-transformer-plugin](https://gatsby-transformer-cloudinary.netlify.com/fluid/)
+
+### Using gatsby-transformer-cloudinary
+
+1. Install `gatsby-transformer-cloudinary` and  `gatsby-source-filesystem` which creates the File nodes that the Cloudinary transformer plugin works on. 
+
+```shell
+ npm install --save gatsby-transformer-cloudinary gatsby-source-filesystem 
+```
+2. In the root of your project, create an environment file called `.env` to which to add your Cloudinary credentials and their values
```
Comment (laurieontech) [inline docs/docs/using-cloudinary-image-service.md:98]:
```suggestion
2. In the root of your project, create an environment file called `.env` to which to add your Cloudinary credentials and their values.
```

File: docs/docs/using-cloudinary-image-service.md [hunk 25]
```diff
@@ -0,0 +1,141 @@
...
+
+> The `info` log, which displays the Cloudinary Media nodes. Those images are ready to be queried in Gatsby components.
+
+## Gatsby-transformer-cloudinary
+
+After sourcing media files from Cloudinary, you will be able to leverage on Cloudinary’s media transformation capabilities you will make use of the `gatsby-transformer-cloudinary` which is a type of [transformer plugin](/docs/creating-a-transformer-plugin/) that is used to change image formats , style and dimension. While also optimizes images for minimal file size alongside high visual quality for an improved user experience and minimal bandwidth.
+
+Here's a [demo site that uses the gatsby-transformer-plugin](https://gatsby-transformer-cloudinary.netlify.com/fluid/)
+
+### Using gatsby-transformer-cloudinary
```
[the remark this answers is not in the document]
Comment (Ekwuno) [reply docs/docs/using-cloudinary-image-service.md:90]: It's the same as the source plugin, with the addition of using the gatsby-source-cloudinary plugin. I didn't want to repeat it.

File: docs/docs/using-cloudinary-image-service.md (hunk 12 above)
Comment (Ekwuno) [reply docs/docs/using-cloudinary-image-service.md:21]: The Cloudinary plugin offers a wider range of transformations than gatsby-images

File: docs/docs/using-cloudinary-image-service.md (hunk 21 above)
Comment (Ekwuno) [reply docs/docs/using-cloudinary-image-service.md:82]: I can take it out

File: docs/docs/using-cloudinary-image-service.md (hunk 25 above)
[the remark this answers is not in the document]
Comment (laurieontech) [reply docs/docs/using-cloudinary-image-service.md:93]: Given that they're under different headers I would repeat it. It gives people the ability to scan to one section or another. Alternatively, you could note it prior to introducing either plugin.

File: docs/docs/using-cloudinary-image-service.md (hunk 12 above)
Comment (laurieontech) [reply docs/docs/using-cloudinary-image-service.md:22]: I'd say that instead. I read over as using the two in combination allowed you to get more out of gatsby-image.

**Comment by laurieontech** [comment]
Answered the two outstanding questions. Let me know when those changes are in and I'll do a final pass :)

File: docs/docs/using-cloudinary-image-service.md (hunk 25 above)
[the remark this answers is not in the document]
Comment (Ekwuno) [reply docs/docs/using-cloudinary-image-service.md:102]: I added a prereq section to this part.

File: docs/docs/using-cloudinary-image-service.md [hunk 26]
```diff
@@ -0,0 +1,156 @@
...
+
+        // This folder will be created if it doesn’t exist.
+        uploadFolder: 'gatsby-cloudinary',
+
+    },
+  ],
+};
+```
+
+> In `gatsby-config.js`, responsive breakpoints can be created for each image, use the fluidMaxWidth and fluidMinWidth options to set them.
```
Comment (laurieontech) [inline docs/docs/using-cloudinary-image-service.md:149]: Last outstanding question I have is whether we have a link for more information? (maybe the README?)

File: docs/docs/using-cloudinary-image-service.md [hunk 27]
```diff
@@ -0,0 +1,156 @@
...
+After sourcing media files from Cloudinary, you will be able to leverage Cloudinary’s media transformation capabilities. To do so, use `gatsby-transformer-cloudinary` which is a type of [transformer plugin](/docs/creating-a-transformer-plugin/) that is used to change image formats, styles and dimensions. It also optimizes images for minimal file size alongside high visual quality for an improved user experience and minimal bandwidth.
+
+Here's a [demo site that uses the gatsby-transformer-plugin](https://gatsby-transformer-cloudinary.netlify.com/fluid/)
+
+### Prerequisites
+
+Before using the `gatsby-transformer-cloudinary` plugin you should do the following:
+
+- Upload your images to a folder on Cloudinary. This folder can have any name of your choosing.
+- Have the gatsby-source-cloudinary plugin to get images.
```
Comment (laurieontech) [inline docs/docs/using-cloudinary-image-service.md:98]:
```suggestion
- Have the `gatsby-source-cloudinary` plugin installed and configured.
```

File: docs/docs/using-cloudinary-image-service.md [hunk 28]
```diff
@@ -146,7 +146,7 @@ module.exports = {
 };
 ```
 
-> In `gatsby-config.js`, responsive breakpoints can be created for each image, use the fluidMaxWidth and fluidMinWidth options to set them.
+> In `gatsby-config.js`, responsive breakpoints can be created for each image, use the fluidMaxWidth and fluidMinWidth options to set them. Here is a [link to the offical documentation](https://www.npmjs.com/package/gatsby-transformer-cloudinary#api) for more information on how these parameters can be set.
```
Comment (laurieontech) [inline docs/docs/using-cloudinary-image-service.md:149]:
```suggestion
> In `gatsby-config.js`, responsive breakpoints can be created for each image, use the fluidMaxWidth and fluidMinWidth options to set them. Take a look at the [plugin documentation](https://www.npmjs.com/package/gatsby-transformer-cloudinary#api) for more information on how these parameters can be set. 
```

File: docs/docs/using-cloudinary-image-service.md (hunk 26 above)
Comment (Ekwuno) [reply docs/docs/using-cloudinary-image-service.md]: Well, these parameters are just to show how to add responsive breakpoints there is no further information after that. I did, however, link the README from npm for context.

**Comment by laurieontech** [comment]
Looks great! LGTM

**Pull request merged**